src/app/state/courses/course.effects.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable } from '@angular/core';
+import { Actions, ofType, createEffect } from '@ngrx/effects';
+import { exhaustMap, map, withLatestFrom } from 'rxjs/operators';
+import { CourseService } from 'src/app/course/course.service';
+import { Store } from '@ngrx/store';
+import { State } from '..';
+import { currentCourse, editCourse, getCoursesSuccess, getSomeCourses} from './courses.actions';
+
+
+@Injectable()
+export class CourseEffects {
+getAllCourses = createEffect(() => 
+  this.actions$.pipe(
+    ofType('Get All Courses'),
```

Comment:
Never use event name directly. It's much difficult to track events and data flow with this approach. 

You have export events, import them in effects and use as a reference.
```typescript
import { getAllCourses} from './courses.actions';

@Injectable()
export class CourseEffects {
getAllCourses = createEffect(() => 
  this.actions$.pipe(
    ofType(getAllCourses),          // <------------
    ...
  )
);

```